lib/carrierwave/neo4j.rb

```diff
@@ -76,12 +152,62 @@ def _set_uploaders_nil
               end
             end
           end
-        end
+
+          def _force_uploaders_reload
+            @_mounters.each do |_, mounter|
+              mounter.send(:uploaders)
+            end
+          end
+
+          # okay, this actually works:
+          def force_retrieve_#{column}
+            send(:#{column}).send(:retrieve_from_store!, #{column}_identifier)
+          end
+
+          # these produce an infinite loop, so... don't reintroduce them:
+          # alias_method :read_uploader, :read_attribute
+          # alias_method :write_uploader, :write_attribute
+          # public :read_uploader
+          # public :write_uploader
+          
+          def read_uploader(name)
+            send(:attribute, name.to_s)
+          end
+
+          def write_uploader(name, value)
+            send(:attribute=, name.to_s, value)
+          end
+
+          def reload_from_database
+            if reloaded = self.class.load_entity(neo_id)
+              self.class.reload_callbacks.each { |m| reloaded.send(m) }
+            end
+            reloaded
+          end
+
+          def reload_from_database!
+            if reloaded = self.class.load_entity(neo_id)
+              send(:attributes=, reloaded.attributes)
+              self.class.reload_callbacks.each { |m| send(m) }
+            end
+            self
+          end
+
+          # MONGOID:
+          # CarrierWave 1.1 references ::ActiveRecord constant directly which
+          # will fail in projects without ActiveRecord. We need to overwrite this
+          # method to avoid it.
+          # See https://github.com/carrierwaveuploader/carrierwave/blob/07dc4d7bd7806ab4b963cf8acbad73d97cdfe74e/lib/carrierwave/mount.rb#L189
+          def store_previous_changes_for_#{column}
+            @_previous_changes_for_#{column} = changes[_mounter(:#{column}).serialization_column]
+          end
+
         RUBY
       end
-    end
 
+
+    end
   end
 end
 
-Neo4j::ActiveNode.send :include, CarrierWave::Neo4j
\ No newline at end of file
+Neo4j::ActiveNode.send :include, CarrierWave::Neo4j
```

Comment:
`Neo4j::ActiveNode.include CarrierWave::Neo4j` Does not work?

Reply:
Unfortunately, no. I didn't alter this line of code (the change is just a carriage return) but I did try switching to an `.include` and it didn't take — so I left it as it was.

I have no idea why. Sorry. I haven't endured Ruby's awful runtime "metaprogramming" for 15 years and I've forgotten most of Hal Fulton's book by now. I actually pinged @cheerfulstoic on Twitter in a moment of desperation but didn't hear back so the semantics of this may remain a mystery forever. ;)

Reply:
okay, I will look into it.

Reply:
Good luck! Be very careful with this, if you do decide to change it. I found that there are situations in which the unit tests do not, for whatever reason, behave identically to a real Rails app so you will need to click-test any changes you make, especially if they affect the meta-programming magic.

Reply:
Okay. Thanks